e2e/playwright/testing-constraints.spec.ts

```diff
@@ -1028,3 +1039,58 @@ part002 = startSketchOn('XZ')
     await expect(page.getByTestId('segment-overlay')).toHaveCount(2)
   })
 })
+testFixture.describe('Electron constraint tests', () => {
+  testFixture(
+    'Able to double click label to set constraint',
+    { tag: '@electron' },
+    async ({ tronApp, homePage, scene, editor, toolbar }) => {
+      await tronApp.initialise({
+        fixtures: { homePage, scene, editor, toolbar },
+        folderSetupFn: async (dir) => {
+          const bracketDir = join(dir, 'test-sample')
+          await fsp.mkdir(bracketDir, { recursive: true })
+          await fsp.copyFile(
+            executorInputPath('angled_line.kcl'),
+            join(bracketDir, 'main.kcl')
+          )
+        },
+      })
+      const [clickHandler] = scene.makeMouseHelpers(600, 300)
+
+      await test.step('setup test', async () => {
+        await homePage.expectState({
+          projectCards: [
+            {
+              title: 'test-sample',
+              fileCount: 1,
+            },
+          ],
+          sortBy: 'last-modified-desc',
+        })
+        await homePage.openProject('test-sample')
+        await scene.waitForExecutionDone()
+      })
+
+      await test.step('Double click to constrain', async () => {
+        await clickHandler()
+        await tronApp.page.getByRole('button', { name: 'Edit Sketch' }).click()
+        const child = tronApp.page
+          .locator('.segment-length-label-text')
+          .first()
+          .locator('xpath=..')
+        await child.dblclick()
+        const cmdBarSubmitButton = tronApp.page.getByRole('button', {
+          name: 'arrow right Continue',
+        })
+        await cmdBarSubmitButton.click()
+        await expectFixture(tronApp.page.locator('.cm-content')).toContainText(
+          'length001 = 15.3'
```

Comment:
Where do you actually set this value? I'm probably reading this wrong

Reply:
When the command bar is opened for setting a constraint the length of the line is the default value to dimension. So the input field is auto populated with this value.

This should always be the same because the initial `KCL` code will be the same so the length of that line will always be the same.

Reply:
Ah right that makes sense, ty!